core/modules/user.py

```diff
@@ -228,47 +228,46 @@ def login(self, name=None, password=None, skey="", *args, **kwargs):
 
         name = name.lower().strip()
         query = db.Query(self.userModule.viewSkel().kindName)
-        res = query.filter("name.idx >=", name).getEntry()
-
-        if res is None:
-            res = {"password": {"pwhash": "-invalid-", "salt": "-invalid-"}, "status": 0, "name": {}}
+        res = query.filter("name.idx >=", name).getEntry() or {}  # might find another user; always keep a dict
 
         password_data = res.get("password") or {}
         # old password hashes used 1001 iterations
         iterations = password_data.get("iterations", 1001)
-        passwd = encode_password(password, password_data.get("salt", ""), iterations)["pwhash"]
+        passwd = encode_password(password, password_data.get("salt", "-invalid-"), iterations)["pwhash"]
 
         # Check if the username matches
-        stored_user_name = (res.get("name") or {}).get("idx", "")
+        stored_user_name = (res.get("name") or {}).get("idx") or ""
```

Comment:
```suggestion
        stored_user_name = user_entry.get("name", {}).get("idx", "")
```

Reply:
No, the or is correct here, because in case the attribute is set but is None (whyever) it should fallback to the default.

So this is my proposal:
```suggestion
        stored_user_name = (user_entry.get("name") or {}).get("idx") or ""
```